CVC5 1.0.8 and 1.0.9 SegFault then non-termination on simple proof

CVC5 1.0.8 exhibits SegFault following by non-termination on the attached SMT file. CVC4 terminates with "unknown".

for example
```
$ cvc4 --rlimit=1000000 seggy.smt2
unknown
```
while
```
$ cvc5 --decision=internal --rlimit=1000000 seggy.smt2
cvc5 suffered a segfault.
Offending address is 0x16d718000
```
and then hangs consuming 100% CPU time.

**Command line arguments**: Command line arguments passed to cvc5 if applicable  
**cvc5 version/commit**: 1.0.8 branch HEAD 
**Operating system**: Apple Silicon macOS 13.5.2 

[seggy.smt2.gz](https://github.com/cvc5/cvc5/files/13442777/seggy.smt2.gz)

I am having trouble building on macOS... I'm getting:
```
rodchap@f4d4889dcf6d build % make
[  0%] Performing build step for 'Poly-EP'
CMake Error at /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-build-Production.cmake:37 (message):
  Command failed: 2

   '/opt/homebrew/bin/gmake' 'poly' 'polyxx'

  See also

    /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-build.log


-- Log output is:
[  2%] Building C object src/CMakeFiles/poly.dir/utils/debug_trace.c.o
In file included from /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP/src/utils/debug_trace.c:20:
/Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP/src/utils/debug_trace.h:22:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^~~~~~~~~
1 error generated.
gmake[6]: *** [src/CMakeFiles/poly.dir/build.make:76: src/CMakeFiles/poly.dir/utils/debug_trace.c.o] Error 1
gmake[5]: *** [CMakeFiles/Makefile2:158: src/CMakeFiles/poly.dir/all] Error 2
gmake[4]: *** [CMakeFiles/Makefile2:165: src/CMakeFiles/poly.dir/rule] Error 2
gmake[3]: *** [Makefile:179: poly] Error 2

CMake Error at /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-build-Production.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [deps/src/Poly-EP-stamp/Poly-EP-build] Error 1
make[1]: *** [CMakeFiles/Poly-EP.dir/all] Error 2
make: *** [all] Error 2
```

Any ideas?

Failure now trying to build cadical:
```
[ 17%] Building CXX object src/CMakeFiles/cvc5-obj.dir/prop/cadical.cpp.o
/Users/rodchap/Desktop/rod/tools/src/cvc5/src/prop/cadical.cpp:68:43: error: expected class name
class CadicalPropagator : public CaDiCaL::ExternalPropagator
                                          ^
/Users/rodchap/Desktop/rod/tools/src/cvc5/src/prop/cadical.cpp:88:50: error: only virtual member functions can be marked 'override'
  void notify_assignment(int lit, bool is_fixed) override
                                                 ^~~~~~~~
```
(and many more errors like that one...all in cadical.cpp)

The compiler is Apple's "gcc" (which is really clang):
```
% gcc --version
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```
rodchap@f4d4889dcf6d build % make clean
rodchap@f4d4889dcf6d build % make -j2
[  0%] Creating directories for 'Poly-EP'
[  0%] Creating directories for 'SymFPU-EP'
[  0%] Performing download step (download, verify and extract) for 'Poly-EP'
[  0%] Performing download step (download, verify and extract) for 'SymFPU-EP'
-- SymFPU-EP download command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-download-*.log
[  0%] No update step for 'SymFPU-EP'
[  0%] No patch step for 'SymFPU-EP'
[  0%] No configure step for 'SymFPU-EP'
-- Poly-EP download command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-download-*.log
[  0%] No update step for 'Poly-EP'
[  0%] No build step for 'SymFPU-EP'
[  0%] Performing patch step for 'Poly-EP'
[  0%] Performing install step for 'SymFPU-EP'
-- Poly-EP patch command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-patch-*.log
-- SymFPU-EP install command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/SymFPU-EP-stamp/SymFPU-EP-install-*.log
[  0%] Performing configure step for 'Poly-EP'
[  1%] Completed 'SymFPU-EP'
[  1%] Built target SymFPU-EP
[  1%] Generating type_enumerator.cpp
-- Poly-EP configure command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-configure-*.log
[  1%] Performing build step for 'Poly-EP'
make[2]: *** read jobs pipe: Resource temporarily unavailable.  Stop.
make[2]: *** Waiting for unfinished jobs....
[  1%] Generating theory_traits.h
make[1]: *** [src/theory/CMakeFiles/gen-theory.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
-- Poly-EP build command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-build-*.log
[  1%] Performing install step for 'Poly-EP'
-- Poly-EP install command succeeded.  See also /Users/rodchap/Desktop/rod/tools/src/cvc5/build/deps/src/Poly-EP-stamp/Poly-EP-install-*.log
[  1%] Performing cleanup step for 'Poly-EP'
[  2%] Completed 'Poly-EP'
[  2%] Built target Poly-EP
make: *** [all] Error 2
```

--- a/cmake/FindGMP.cmake
+++ b/cmake/FindGMP.cmake
@@ -70,7 +70,7 @@ if(NOT GMP_FOUND_SYSTEM)
 
   include(ExternalProject)
 
-  set(GMP_VERSION "6.2.1")
+  set(GMP_VERSION "6.3.0")
 
   set(GMP_INCLUDE_DIR "${DEPS_BASE}/include/")
   if(BUILD_SHARED_LIBS)
@@ -123,7 +123,7 @@ if(NOT GMP_FOUND_SYSTEM)
     GMP-EP
     ${COMMON_EP_CONFIG}
     URL https://ftp.gnu.org/gnu/gmp/gmp-${GMP_VERSION}.tar.bz2
-    URL_HASH SHA1=2dcf34d4a432dbe6cce1475a835d20fe44f75822
+    URL_HASH SHA1=32d21c4fae046de45e8fce37bf4002236d283b71
     CONFIGURE_COMMAND
       ${CONFIGURE_ENV}
           ${CONFIGURE_CMD_WRAPPER} ${SHELL} <SOURCE_DIR>/configure

